presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -270,11 +258,20 @@ private PrestoSparkQueryExecution(
             queryCompletedEvent(Optional.empty());
 
             ConnectorSession connectorSession = session.toConnectorSession();
-            return resultRdd.stream()
-                    .map(Tuple2::_2)
-                    .map(this::deserializePage)
-                    .flatMap(page -> getPageValues(connectorSession, page, rootFragment.getTypes()).stream())
-                    .collect(toList());
+            List<Type> types = rootFragment.getTypes();
+            ImmutableList.Builder<List<Object>> result = ImmutableList.builder();
+            for (Tuple2<Integer, PrestoSparkRow> tuple : resultRdd) {
+                PrestoSparkRow row = tuple._2;
+                SliceInput sliceInput = new BasicSliceInput(Slices.wrappedBuffer(row.getBytes(), 0, row.getLength()));
+                ImmutableList.Builder<Object> columns = ImmutableList.builder();
+                for (Type type : types) {
```

Comment:
I am wondering if we should have a utility function to convert a list of `PrestoSparkRow` into a list of Blocks? -- we can leave this as TODO .

Reply:
Currently this is a single place where it is done. Let's keep it here for now.